tcping.go

```diff
@@ -526,21 +522,20 @@ func (tcpStats *stats) handleConnSuccess(rtt float32, now time.Time) {
 func tcping(tcpStats *stats) {
 	IPAndPort := netip.AddrPortFrom(tcpStats.ip, tcpStats.port)
 
-	connStart := getSystemTime()
-	conn, err := net.DialTimeout("tcp", IPAndPort.String(), oneSecond)
+	connStart := time.Now()
+	conn, err := net.DialTimeout("tcp", IPAndPort.String(), time.Second)
 	connEnd := time.Since(connStart)
 
 	rtt := nanoToMillisecond(connEnd.Nanoseconds())
-	now := getSystemTime()
 
 	if err != nil {
-		tcpStats.handleConnError(now)
+		tcpStats.handleConnError(time.Now())
 	} else {
-		tcpStats.handleConnSuccess(rtt, now)
+		tcpStats.handleConnSuccess(rtt, time.Now())
```

Comment:
Using the previous method, calling time.now() would result in some incorrect/inaccurate time reports.

That is why I was getting the system time before checking the conditions.

Using your method, have you noticed any odd behaviors or reports seem fine? Apart from the bug we discussed.

Reply:
Oh, I see now. How about we use `connStart` instead of time.Now() inside conditions? It shouldn't be adding any extra time that way, since It's 100% assigned to time.Now() and will have the same time, regardless of `if`

Reply:
Initially I had gone that route (using `connStart`) but I noticed depending on how fast or slow a connection establishes or fails, it also causes crazy drifts in the long run. Meaning when the program is running for hours and days.

That is why I just resorted to not using `connStart` for later calculations.

Reply:
Of course, we could really benefit from a proper test case that could bring this to light and not having it run for hours/days but I have no idea how to address  that myself.